src/OT/Color/COLR/COLR.hh

```diff
@@ -242,7 +251,11 @@ struct Variable
   }
 
   void closurev1 (hb_colrv1_closure_context_t* c) const
-  { value.closurev1 (c); }
+  {
+    // update c->num_var_idxes during value closure
+    value.closurev1 (c);
```

Comment:
Can you init c->num_var_idxes to 0 before calling the closure. If the method doesn't happen to set a value for it then add_var_idxes below will just use whatever the previously set value is.